spec/SchemaPerformance.spec.js

```diff
@@ -0,0 +1,212 @@
+const Config = require('../lib/Config');
+const MongoStorageAdapter = require('../lib/Adapters/Storage/Mongo/MongoStorageAdapter').default;
+const mongoURI = 'mongodb://localhost:27017/parseServerMongoAdapterTestDatabase';
+
+describe_only_db('mongo')('Schema Performance', function () {
+  let getAllSpy;
+  let config;
+
+  beforeEach(async () => {
+    config = Config.get('test');
+    config.schemaCache.clear();
+    const databaseAdapter = new MongoStorageAdapter({ uri: mongoURI });
+    await reconfigureServer({
+      horizontalScaling: false,
+      databaseAdapter,
+    });
+    getAllSpy = spyOn(databaseAdapter, 'getAllClasses').and.callThrough();
+  });
+
+  it('test new object', async () => {
+    const object = new TestObject();
+    object.set('foo', 'bar');
+    await object.save();
+    expect(getAllSpy.calls.count()).toBe(2);
+  });
+
+  it('test new object multiple fields', async () => {
+    const container = new Container({
+      dateField: new Date(),
+      arrayField: [],
+      numberField: 1,
+      stringField: 'hello',
+      booleanField: true,
+    });
+    await container.save();
+    expect(getAllSpy.calls.count()).toBe(2);
+  });
+
+  it('test update existing fields', async () => {
+    const object = new TestObject();
+    object.set('foo', 'bar');
+    await object.save();
+
+    getAllSpy.calls.reset();
+
+    object.set('foo', 'barz');
+    await object.save();
+    expect(getAllSpy.calls.count()).toBe(0);
+  });
+
+  xit('test saveAll / destroyAll', async () => {
```

Comment:
This is on purpose. I left a comment explaining why. In case I need to check performance on 10,000 objects. I don't have to write it. Actually I don't think its flaky anymore with the recent test updates.

Reply:
Ok